File: ATaleOfThreeCities/solve/python/ATaleOfThreeCitiesSolver.py
#! /usr/bin/env python
import os, sys
import ATaleOfThreeCities

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        ax = reader.next("int[]")
        reader.next()
        ay = reader.next("int[]")
        reader.next()
        bx = reader.next("int[]")
        reader.next()
        by = reader.next("int[]")
        reader.next()
        cx = reader.next("int[]")
        reader.next()
        cy = reader.next("int[]")

    result = ATaleOfThreeCities.connect(ax, ay, bx, by, cx, cy)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "double"))

if __name__ == "__main__":
    init()
    main()

File: ATaleOfThreeCities/solve/javascript/ATaleOfThreeCitiesSolver.js
#! /usr/bin/env node

var fs = require("fs");
var path = require("path");
var tc = null;

function init() {
  var gettcHome = process.env.GETTC_HOME || path.join(process.env.HOME, ".gettc");
  var includeDir = path.join(gettcHome, "include", "javascript");
  tc = require(path.join(includeDir, "topcoder"));
}

function main() {
  try {
    var input = fs.readFileSync(process.argv[2], { encoding: "ascii" });
    var reader = new tc.Reader(input);
        var ax = reader.next('int[]'); reader.next();
    var ay = reader.next('int[]'); reader.next();
    var bx = reader.next('int[]'); reader.next();
    var by = reader.next('int[]'); reader.next();
    var cx = reader.next('int[]'); reader.next();
    var cy = reader.next('int[]');

    var ATaleOfThreeCities = require("./ATaleOfThreeCities");
    var result = ATaleOfThreeCities.connect(ax, ay, bx, by, cx, cy);

    var writer = new tc.Writer();
    writer.next(result, "double");
    fs.writeFileSync(process.argv[3], writer.toString(), { encoding: "ascii" });
  } catch (err) {
    console.log(err.toString());
    console.trace();
  }
}

init();
main();
